### PR TITLE
Classed state config

### PR DESCRIPTION
**Pull request by ceesem**
Large update that focuses on the statebuilder.
1. It no longer imports neuroglancer to avoid fork issues with the google mainline.
2. Statebuilder is configured almost completely differently, but it should be much more simple now. Each layer has its own config that is well documented in docstrings and can be explored more easily through tag autocompletes.
3. Statebuilder works now with the main Seung lab branch of Neuroglancer.
4. Examples of direct use and use with dashdataframes are in /examples

### CODE REVIEW

**Comment by ceesem** [comment]
5. Tests even work again!

**Comment by ceesem** [comment]
Oh, key caveat! Because of the oncoming need for Auth to handle the state server and info server requests, this needs the updated framework client we had before. We should do a PR on that, bump the version, and set the requirements here first.

File: examples/dash_with_statebuilder.py [hunk 1]
```diff
@@ -9,37 +9,47 @@
...
 def build_dash_app():
     df = pd.read_hdf(df_filename, df_key)
 
     app = dash.Dash()
 
-    sb = StateBuilder(dataset_name='pinky100',
-                      selected_ids={'seg': ['pt_root_id']},
-                      point_annotations={'soma': ['pt_position']})
-    
+    sb = _statebuilder()
```
Comment (fcollman) [inline examples/dash_with_statebuilder.py:38]: isnt' this _ a typo?

File: src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+# limitations under the License.
+
+from . import credentials_provider, google_credentials
+
+default_credentials_manager = credentials_provider.CredentialsManager()
+default_credentials_manager.register(
+    u'google-brainmaps',
+    lambda _parameters: google_credentials.GoogleCredentialsProvider(
+        client_id=u'639403125587-ue3c18dalqidqehs1n1p5rjvgni5f7qu.apps.googleusercontent.com',
+        client_secret=u'kuaqECaVXOKEJ2L6ifZu4Aqt',
```
Comment (fcollman) [inline src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py:22]: should this be in here? Whose credentials are these?

**Comment by fcollman** [comment]
This looks awesome to me.

File: src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py (hunk 2 above)
Comment (ceesem) [reply src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py:22]: Good point. That was in the original neuroglancer module and I was trying to keep a light touch for parts of the code when I wasn't sure precisely what it was going to be useful for. Now that I'm doing code coverage, though, it's clear I can get rid of this and a few other files for the intended use case.

File: examples/dash_with_statebuilder.py (hunk 1 above)
Comment (ceesem) [reply examples/dash_with_statebuilder.py:38]: Not a typo, but less clear than it should be. Changed.

File: src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py (hunk 2 above)
Comment (ceesem) [reply src/neuroglancer_annotation_ui/nglite/default_credentials_manager.py]: I removed several files from the reduced neuroglancer package

**Pull request merged**